Auth controller

Drunken-Dragon opened the pull request:
Aplikacja zwraca błąd 'Identifier "form" is not defined.'

Przyczyna błędu umyka mi już o tej porze...Proszę o review.

Drunken-Dragon committed:
added login verification to AuthController

Drunken-Dragon committed:
created LoginFormController

nexik commented on src/app.php:
```diff
@@ -50,7 +50,7 @@
 
 $app['auth.controller'] = function () use ($app) {
-    return new \Controller\AuthController($app['twig']);
+    return new \Controller\AuthController($app['twig'], $app['form'], new \Controller\LoginFormController($app['form.factory']));
```
constructor przyjmuje tylko 2 argumenty a ty przekazujesz 3

nie masz zdefiniowane $app['form'] wiec silex nie wie co to jest 'form'

musisz zdefiniowac $app['form'] = $app['form.factory']->createBuilder(FormType::class, $data) ....

Drunken-Dragon committed:
created displayForm method

Drunken-Dragon committed:
reestablished connection with the db

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,23 +2,69 @@
-    public function __construct($twig, $form)
+    private $db;
+
+    public function __construct($app, $db)
```
jak nie korzystasz z $db to nie wymagaj tego w construct

nexik commented on src/app.php:
```diff
@@ -50,7 +50,7 @@
 
 $app['auth.controller'] = function () use ($app) {
-    return new \Controller\AuthController($app['twig'], $app['form'], new \Controller\LoginFormController($app['form.factory']));
+    return new \Controller\AuthController($app, $db);
```
zmienna $db nie jest zdefiniowana.

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,23 +2,69 @@
+
+    public function loginAction(Request $request)
+    {
+
```
nie potrzbna pusta linia

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,23 +2,69 @@
+            $user = $db->fetchAll('SELECT * FROM users WHERE name = ?', [$data['name']]);
 
             if (password_verify($data['password'], $user[0]['password'])) {
+                dump($app['session']);
```
takie rzeczy mozna lokalnie a ale nie kodzie "produkcyjnym" w  PR tak samo ten die()

nexik commented on src/Controller/LoginFormController.php:
```diff
@@ -15,8 +15,8 @@ class LoginFormController
     {
-        $this->form = $form;
-        $this->app = $app;
+//        $this->form = $formFactory->;
```
usuwamy nie komentujemy kodu

nexik commented on src/Controller/LoginFormController.php:
```diff
@@ -15,8 +15,8 @@ class LoginFormController
-        $this->form = $form;
-        $this->app = $app;
+//        $this->form = $formFactory->;
+//        $this->app = $app;
```
usuwamy nie komentujemy kodu

Drunken-Dragon commented:
Nie skonczylem jeszcze tego etapu,stad sporo komentarzy. Dzisiaj chce dojsc
do etapu kiedy sorawdza sesje i wczytuje lp. Wtedy uporzadkuje kod przed
kolejnym pushem.


09.12.2017 13:37 "Tomasz Ślązok" <notifications@github.com> napisał(a):

*@nexik* requested changes on this pull request.
------------------------------

In src/Controller/AuthController.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921658>:

>      private $twig;
-    public function __construct($twig, $form)
+    private $db;
+
+    public function __construct($app, $db)

jak nie korzystasz z $db to nie wymagaj tego w construct
------------------------------

In src/app.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921661>:

> @@ -50,7 +50,7 @@
 });

 $app['auth.controller'] = function () use ($app) {
-    return new \Controller\AuthController($app['twig'], $app['form'],
new \Controller\LoginFormController($app['form.factory']));
+    return new \Controller\AuthController($app, $db);

zmienna $db nie jest zdefiniowana.
------------------------------

In src/Controller/AuthController.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921670>:

>      {
-        $form->handleRequest($request);
+        return $this->twig->render('login.html.twig', [
+            'form' => $this->form->createView()
+        ]);
+    }
+
+    public function loginAction(Request $request)
+    {
+

nie potrzbna pusta linia
------------------------------

In src/Controller/AuthController.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921682>:

>
             if (password_verify($data['password'], $user[0]['password'])) {
+                dump($app['session']);

takie rzeczy mozna lokalnie a ale nie kodzie "produkcyjnym" w PR tak samo
ten die()
------------------------------

In src/Controller/LoginFormController.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921695>:

> @@ -15,8 +15,8 @@ class LoginFormController

     public function __construct($formFactory)
     {
-        $this->form = $form;
-        $this->app = $app;
+//        $this->form = $formFactory->;

usuwamy nie komentujemy kodu
------------------------------

In src/Controller/LoginFormController.php
<https://github.com/Drunken-Dragon/silex-lp/pull/6#discussion_r155921699>:

> @@ -15,8 +15,8 @@ class LoginFormController

     public function __construct($formFactory)
     {
-        $this->form = $form;
-        $this->app = $app;
+//        $this->form = $formFactory->;
+//        $this->app = $app;

usuwamy nie komentujemy kodu

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Drunken-Dragon/silex-lp/pull/6#pullrequestreview-82324601>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AWj0bKbxta7FmafiBNwIPbC4LXYC_ULMks5s-n75gaJpZM4Q4xia>
.

nexik commented:
ok spoko tak wiec bede czekac na part II :) milego weekendu.

Drunken-Dragon committed:
refactored variables used in LandingController

Drunken-Dragon committed:
Refactored variables in LoginFormController

Drunken-Dragon committed:
-moved login form rendering to AuthController
-moved login form validation to AuthController
-renamed controllers.php as routes.php

Drunken-Dragon commented:
Part II ready :) Pora nieco się przespać...

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,15 +2,76 @@
+
+    public function __construct($app, $db, $session)
+    {
+        $this->db = $app['db'];
```
Albo przekazujesz do constructora $db albo uzywasz $app['db'] musisz sie na cos zdecydowac.

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,15 +2,76 @@
+    public function __construct($app, $db, $session)
+    {
+        $this->db = $app['db'];
+        $this->session = $app['session'];
```
Albo przekazujesz do constructora $session albo uzywasz $app['session'] musisz sie na cos zdecydowac.

nexik commented on src/Controller/LandingController.php:
```diff
@@ -11,8 +11,8 @@ class LandingController
         $this->app = $app;
-        $this->session = $session;
-        $this->twig = $twig;
+        $this->session = $app['session'];
```
Albo przekazujesz do constructora $session albo uzywasz $app['session'] musisz sie na cos zdecydowac.

nexik commented on src/Controller/LandingController.php:
```diff
@@ -11,8 +11,8 @@ class LandingController
-        $this->session = $session;
-        $this->twig = $twig;
+        $this->session = $app['session'];
+        $this->twig = $app['twig'];
```
Albo przekazujesz do constructora $twig albo uzywasz $app['twig'] musisz sie na cos zdecydowac.

nexik commented on src/routes.php:
```diff
@@ -0,0 +1,31 @@
+use Symfony\Component\Validator\Constraints as Assert;
+
+$app->get('/', 'landing.controller:verifyAccess');
+
```
puste linie podziedzy wowolaniami $app->get() mozna usuwac nie poprawiaja znaczaco czytelnosci kodu aby je dodawac

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,15 +2,76 @@
+
+        if ($this->form->isValid()) {
+            $data = $this->form->getData();
+
```
ta pusta linia mozna usuwac nic nie wnosi

nexik commented on src/Controller/AuthController.php:
```diff
@@ -2,15 +2,76 @@
+                return $app->redirect('/login');
+            }
+        }
+        return $this->twig->render('login.html.twig', array('form' => $form->createView()));
```
przed returnem dodajemy pusta linie aby bylo widac ze to jest najwazniejsza czesc metody

Drunken-Dragon committed:
Refactored variables in AuthController and LandingController.

Drunken-Dragon committed:
Added .env and used DB credentials from .env file.

Drunken-Dragon commented:
Wprowadziłem Twoje poprawki oraz dodałem .env.

Wydaje mi się, że skorzystałem z .env poprawnie, jednak rzuca mi błąd:
An exception occured in driver: SQLSTATE[HY000] [1045] Access denied for user 'root'@'localhost' (using password: NO)

Czy możesz jutro na to zerknąć? Póki co nie znalazłem w necie rozwiązania które by mi pomogło...

nexik commented:
jedyny blad jaki widze to brak zaladowanie .dotenva w projekcie .env sam sie nie zaladuje

Drunken-Dragon commented:
czy to wymaga użycia jakiegoś osobnego service?

2017-12-10 18:42 GMT+01:00 Tomasz Ślązok <notifications@github.com>:

> jedyny blad jaki widze to brak zaladowanie .dotenva w projekcie .env sam
> sie nie zaladuje
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Drunken-Dragon/silex-lp/pull/6#issuecomment-350564946>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AWj0bLyOAt8RAvnFeCWOuNjmBye9Bwrjks5s_Bf9gaJpZM4Q4xia>
> .
>

nexik commented:
musisz zaladowac .env jakos: 

mozna na poziomie systemu operacyjnego np. w linuxie:

```
source .env
```

Tak to sie robi na produkcji w wiekszosci przypadkow. Przy srodowisku developerskim mozna skorzystac z https://symfony.com/doc/current/components/dotenv.html

Drunken-Dragon commented:
Nowy PR https://github.com/Drunken-Dragon/silex-lp/pull/8

W dniu 10 grudnia 2017 19:17 użytkownik Tomasz Ślązok <
notifications@github.com> napisał:

> musisz zaladowac .env jakos:
>
> mozna na poziomie systemu operacyjnego np. w linuxie:
>
> source .env
>
> Tak to sie robi na produkcji w wiekszosci przypadkow. Przy srodowisku
> developerskiego mozna skorzystac z https://symfony.com/doc/
> current/components/dotenv.html
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Drunken-Dragon/silex-lp/pull/6#issuecomment-350567324>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AWj0bF_ack0xRWC9FrQSgVWJ_29zE2naks5s_CAggaJpZM4Q4xia>
> .
>
